### ISSUE TITLE
No docs on Plugin API

### ISSUE THREAD

**Issue by BebeSparkelSparkel** [karma-runner/karma#2894]
At http://karma-runner.github.io/1.0/dev/plugins.html there are no docs on the api.

Seems like each should list what is required instead of just listing examples.

From [karma-coffee-preprocessor](https://github.com/karma-runner/karma-coffee-preprocessor/blob/master/index.js) what is:
line 5 what are all the objects passed in as arguments and how are they selected
`$inject` line 51
`'config.coffeePreprocessor'` line 51 (how would I know to do that)
... line 51
`'preprocessor:coffee'` line 55
`'factory'` line 55 (what options are there?)
Basically need to how the initializer works needs to be defined

Then the function returned needs docs.

**Comment by EzraBrooks** [comment]
Duplicate of #2850. Please discuss plugin API docs there.
